editor_drafts/draft-prorock-spice-cose-sd-cwt.md

```diff
@@ -142,10 +142,81 @@ However the following guidance is generally recommended, regardless of protocol
 
 # SD-CWT Issuance
 
+SD-CWT is modeled after SD-JWT, with adjustments to align with conventions in CBOR and COSE.
+
+An SD-CWT is a CWT containing the hash digest (the "blinded claim hash") of each blinded claim in `redacted_values` in the payload, and optionally the salted claim values (and often claim names) for the values that are actually disclosed in the `sd_claims` claim in the unprotected header. When blinding an individual item in an array, the value of the item is replaced with a dict containing only the special key "...".
+
+``` cddl
+redacted_element = { "...": bstr }
+```
+
+A Holder key binding CWT (#kbt) MUST be present in a `sd_kbt` claim in the unprotected header when presenting an SD-CWT to a Verifier.
+The `sd_kbt` claim can only be absent when the Issuer is providing the
+SD-CWT to the Holder.
+
 An SD-CWT is a CWT containing zero or more Digested Salted Disclosed Claim, and zero or more Salted Disclosed Claims.
 The salt acts as a blinding factor, preventing a Verifier of an SD-CWT from learning claims that were not intentionally disclosed by a Holder.
-A confirmation claim `cnf (8)` MUST be present in the CWT Claimset.
-The `sd_kbt` MUST NOT be set by the Issuer, and MUST be set by the Holder, and is therefore marked optional in the following normative defintion of SD-CWT in CDDL:
+
+The following informative CDDL is provided to explain the syntax for an
+SD-CWT presentation. A complete CDDL schema is in (#cddl).
+
+Please note this example contains claims for demonstration of the disclosure syntax, such as `swversion`, `address`, and ``.
+
+``` cddl
+sd-cwt-presentation = #6.18([
+   protected: bstr .cbor sd-protected,
+   unprotected-presentation,
+   payload: bstr .cbor sd-payload,
+   signature: bstr
+])
+
+sd-protected = {
+   &(typ: 16) ^ => "application/sd+cwt",
+   &(alg: 1) ^ => int,
+   * key => any
+}
+
+unprotected-presentation = {
+   &(sd_kbt: TBD2) ^ => bstr .cbor kbt-cwt,
+   ? &(sd_claims: TBD1) ^ => [ +bstr .cbor salted ],
```

Comment:
```suggestion
   ? &(sd_claims: TBD1) ^ => bstr .cbor [ + salted ],
```

Reply:
> In SD-JWT, the sd_claims that are disclosed are hashed from their encoded state. In SD-CWT, we should plan to do the same, otherwise an intermediary do further redaction. I think its better to encode the entire claimset, so the hash is over bytes that are not subject to reordering, from arrays.

I proposed an edn example instance document in the previous commit. This schema validates that instance document, so what we do we have to do to the other.

The disclosures are disclosable individually. So we need at least `[ +bstr .cbor salted ]`. I thought CBOR array were not allowed to be reordered (only maps keys)? If you are really sure the array could get reordered you could also do this:

`? &(sd_claims: TBD1) ^ => bstr .cbor [ +bstr .cbor salted ],

Reply:
Yes, you are right... we need both.

The outer encoding is to quickly verify the key binding token is for "these specific disclosures".

The inner encodings are to construct the intermedia disclosure hash map, to process the cbor map and produce the verified claimset.